### PR TITLE
Adding type specs to emysql.hrl

### PR DESCRIPTION
**Pull request by jkingsbery**
Started work on #125, added type information to the include file.

### CODE REVIEW

File: src/emysql_tcp.erl [hunk 1]
```diff
@@ -110,7 +110,7 @@ parse_response(_Sock, _Timeout, #packet{seq_num = SeqNum, data = <<255:8, ErrNo:
     { #error_packet{
         seq_num = SeqNum,
         code = ErrNo,
-        status = 0,
```
Comment (jlouis) [inline src/emysql_tcp.erl:113]: I guess this change fixes a problem with the wrong return type, correct? As in, it is binary in a lot of other places, but not in this particular case?

**Comment by jlouis** [comment]
I really like these changes, and I will take them. The only two questions I have is the already referred one about the binary-type in the `#error_packet{}` and your rationale for this (It is a valid answer to say the code is crap and it is not type-correct in that case, heh). And why you add a couple of whitespace changes which I can't understand the necessity of.

Otherwise, it LGTM. It should definitely go in!

**Comment by jkingsbery** [comment]
I added a new commit to fix the white space (those changes were from me tinkering with things, not putting them back, and not double checking my commits thoroughly).

The change to binary you noted was required to get the types correct on the `error_packet` record, and it was the simplest change to get it to work. I'm still not too familiar with the emysql_tcp code, so I didn't want to muck with it too much. Ideally, the status codes for all the packet records should have the same type.

**Comment by jlouis** [comment]
Agreed! I'll push this in. Thank you very much!

**Pull request merged**